File: lib/commands/builtin/activate.ts
import type { CommandSpec } from "../types"

export const activateCommand: CommandSpec = {
  name: "activate",
  aliases: ["a"],
  summary: "focus a tab and its window",
  usage: ["activate <tabId>", "a <tabId>"],
  man: [
    "NAME",
    "  activate, a - focus a tab and its window",
    "",
    "SYNOPSIS",
    "  activate <tabId>",
    "  a <tabId>"
  ],
  async execute(_ctx, args) {
    const id = Number(args[1])
    if (!Number.isInteger(id)) {
      return ["usage: activate <tabId>"]
    }
    const tab = await chrome.tabs.get(id)
    await chrome.tabs.update(id, { active: true })
    await chrome.windows.update(tab.windowId, { focused: true })
    return [`activated tab ${id} (window ${tab.windowId})`]
  }
}

File: wasm/bmxt-core/src/cmd/activate.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "activate",
    aliases: &["a"],
    usage_primary: "activate <tabId>",
    man: &[
        "NAME",
        "  activate, a - focus a tab and its window",
        "",
        "SYNOPSIS",
        "  activate <tabId>",
        "  a <tabId>",
    ],
};

pub fn run(args: &[String]) -> DispatchJson {
    let Some(id) = args.get(1).and_then(|s| s.parse::<i32>().ok()) else {
        return DispatchJson::lines(vec!["usage: activate <tabId>".to_string()]);
    };
    DispatchJson::effects(vec![Effect::Activate { tab_id: id }])
}
